src/server.rs

```diff
@@ -36,6 +38,65 @@ pub struct PayloadTraceContext {
     local_to_external_payload_ids: Arc<Mutex<LruCache<PayloadId, PayloadId>>>,
 }
 
+enum PayloadVersion {
+    V3,
+    V4(Vec<Bytes>),
```

Comment:
instead of the payload version encapsulating the execution requests for the v4 new payload request, why not have a structure that encapsulates all the args for the new_payload method as an enum

Reply:
can you elaborate on this?

Reply:
`OpExecutionenvelopeV(3/4)` is from Alloy and fields are not similar to calls of `new_payload` but followed another approach in 4eb5e5187722ecaa7c7e43bff7e74f3eaa429738